tools/serverpod_cli/lib/src/generator/dart/library_generators/library_generator.dart

```diff
@@ -837,6 +839,15 @@ class LibraryGenerator {
     return methodConnectors;
   }
 
+  String? _getMethodCallComment(MethodCallDefinition m) {
+    for (var a in m.annotations) {
+      if (a.methodCallAnalyzerIgnoreRule != null) {
+        return '\n// ignore: ${a.methodCallAnalyzerIgnoreRule}\n';
```

Comment:
Suggestion(Non blocking and opinionated): Use a list of strings for the ignore rules and change the name to method call comments.

With the current solution we place half of the statement in the analyzer (`deprecated_member_use_from_same_package`) and half of the statement in the generator (`ignore:`). A suggestion is to have the full statement in the analyzer and then just have a list of comments that are inserted at the method call site.

That way, you can easily see the fully constructed comment in a single place.

Reply:
That is true, it would be more of the needed logic for this use case is in one place. However it would mean that the analyzer code knows about code generation - e.g. shall an ignore statement be made for a line of for a file. I thought it better for the analyzer to represent the analyzer rule that is relevant, and the generator to decide how to generate ignore code.